src/OrchardCore/OrchardCore.Email.Core/Services/SmtpService.cs

```diff
@@ -34,22 +43,38 @@ IStringLocalizer<SmtpService> stringLocalizer
             S = stringLocalizer;
         }
 
+        /// <summary>
+        /// Sends the specified message to an SMTP server for delivery.
+        /// </summary>
+        /// <param name="message">The message to be sent.</param>
+        /// <returns>A <see cref="SmtpResult"/> that holds information about the sent message, for instance if it has sent successfully or if it has failed.</returns>
+        /// <remarks>This method allows to send an email without setting <see cref="MailMessage.To"/> if <see cref="MailMessage.Cc"/> or <see cref="MailMessage.Bcc"/> is provided.</remarks>
         public async Task<SmtpResult> SendAsync(MailMessage message)
         {
-            if (_options?.DefaultSender == null)
+            if (_options == null)
             {
                 return SmtpResult.Failed(S["SMTP settings must be configured before an email can be sent."]);
             }
 
             try
             {
                 // Set the MailMessage.From, to avoid the confusion between _options.DefaultSender (Author) and submitter (Sender)
-                message.From = String.IsNullOrWhiteSpace(message.From)
+                var senderAddress = String.IsNullOrWhiteSpace(message.From)
                     ? _options.DefaultSender
                     : message.From;
 
+                if (!String.IsNullOrWhiteSpace(senderAddress))
+                {
+                    message.From = senderAddress;
+                }
+
                 var mimeMessage = FromMailMessage(message);
 
+                if (mimeMessage.From.Count == 0 && mimeMessage.Cc.Count == 0 && mimeMessage.Bcc.Count == 0)
+                {
+                    return SmtpResult.Failed(S["The mail message should have at least one of these headers: To, Cc or Bcc."]);
```

Comment:
I'm asking myself if the Mailkit library now won't return proper error messages when we update it to the latest version like you saw here https://github.com/OrchardCMS/OrchardCore/pull/10863